icu4c/source/i18n/unicode/coleitr.h

```diff
@@ -322,6 +335,19 @@ class U_I18N_API CollationElementIterator U_FINAL : public UObject {
     */
     CollationElementIterator(const UnicodeString& sourceText,
         const RuleBasedCollator* order, UErrorCode& status);
+
+   /**
+    * CollationElementIterator read-only alias constructor. Same as the above constructor,
+    * but if readOnly is true then it uses a read-only alias of the sourceText string.
+    * The input source string must outlive this object's lifetime.
+    * @param sourceText    the source string.
+    * @param readOnly      if true, uses a read-only alias of sourceText.
+    * @param order         the collation object.
+    * @param status        the error code status.
+    */
+    CollationElementIterator(const UnicodeString& sourceText, UBool readOnly,
```

Comment:
Why not just add the readOnly parameter to the existing constructor? Too many call sites to modify? Might be a good opportunity to review them to see which needs what...

Another possibility is to not change the constructor nor add a new one at all. Instead, call sites could construct with an empty string and then call the new setter.